ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp

```diff
@@ -217,6 +217,21 @@ Tensor BinaryOperation<binary_op_type, in_place>::operator()(
         input_tensor_a, scalar_tensor_device, dtype, memory_config, optional_output_tensor, activations, input_tensor_a_activation);
 }
 
+template <BinaryOpType binary_op_type, bool in_place>
+ComplexTensor BinaryOperation<binary_op_type, in_place>::operator()(
```

Comment:
What are we doing with this template argument here?

Reply:
Existing ttnn.add api contains inplace flag as template so we used the same here

Reply:
It seems we have a conflicting approach in our api in that an in_place template arg is taking the place of optional_output_tensors.   Leave it as-is on ComplexTensor setup as well for now but we may need to revisit.

Reply:
@Aswinmcw , take a look at what was recently done by @mouliraj-mcw for the Inplace ops.  This is what I would like done for these versions as well.

Reply:
Yes I will take care of those changes